File: Src/Client/src/app/shared/models/time-sheet-report-detail-model.ts
import { TimeSpan } from './timepan';

export class TimeSheetReportDetailModel {
  userId: number;
  firstName: string;
  lastName: string;

  backlogId: string;
  taskId: string;
  tsAcitivityGroupId: string;
  tsActivityId: string;
  tsAcitivityGroupName: string;
  tSActivityName: string;
  timeSheetRecordId: string;
  timeSheetRecordName: string;

  startDate: Date;
  startHour: string;
  endDate: Date;
  endHour: string;
  dateRecord: string;

  nameTask: string;
  duration: TimeSpan;
  totalTime: TimeSpan;
  fullName: string;

  constructor(init?: Partial<TimeSheetReportDetailModel>) {
    Object.assign(this, init);
  }
}

File: Src/Server/Kloon.EmployeePerformance.Models/TimeSheet/TimeSheetReport/TimeSheetReportDetailModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Kloon.EmployeePerformance.Models.TimeSheet.TimeSheetReport
{
    public class TimeSheetReportDetailModel
    {
        public int UserId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }

        public string BacklogId { get; set; }
        public string TaskId { get; set; }
        public Guid TSAcitivityGroupId { get; set; }
        public Guid TSActivityId { get; set; }
        public string TSAcitivityGroupName { get; set; }
        public string TSActivityName { get; set; }

        public Guid TimeSheetRecordId { get; set; }
        public string TimeSheetRecordName { get; set; }

        public DateTime StartDate { get; set; }
        public string StartHour { get; set; }
        public DateTime EndDate { get; set; }
        public string EndHour { get; set; }
        public string DateRecord { get; set; }

        public TimeSpan TotalTime { get; set; }

        public string NameTask
        {
            get
            {
                return TSAcitivityGroupName + ": " + TSActivityName;
            }
        }

        public TimeSpan Duration
        {
            get
            {
                return EndDate - StartDate;
            }
        }

        public string FullName
        {
            get
            {
                return FirstName + " " + LastName;
            }
        }
    }
}
